### PR TITLE
Add dependencies for protobuf and remove third_party forlder

### LINKED ISSUE #1023: BSR - Failure: repository has no commits
There's an error when i try to fetch wasmd's protobuf from `buf.build` like this
``` yaml
# buf.yaml
deps:
  - buf.build/cosmwasm/wasmd
```

``` bash
$ buf mod update
Failure: repository has no commits
```

plus. seems like actions for publish keeps aborted due to dependency issue.
https://github.com/CosmWasm/wasmd/actions/runs/3107091489/jobs/5034779022

### PR DESCRIPTION
**Pull request by pinosu**
Resolves #1023

### CODE REVIEW

File: proto/buf.yaml [hunk 1]
```diff
@@ -5,6 +5,10 @@
 # the previous configuration.
 version: v1
 name: buf.build/cosmwasm/wasmd
+deps:
+  - buf.build/cosmos/cosmos-proto
+  - buf.build/cosmos/cosmos-sdk
```
[the remark this answers is not in the document]
Comment (pinosu) [reply proto/buf.yaml:10]: I would like to use v0.45.8 but it is hard to find a commit that matches a specific tag. Do you have any idea how to do it? I locally run `buf build` and I didn't get any error.

[the remark this answers is not in the document]
Comment (julienrbrt) [reply proto/buf.yaml:10]: We will maintain a table of buf commits per sdk versions: https://github.com/cosmos/cosmos-sdk/blob/main/third_party/proto/README.md. However, due to #1415 we only have for `v0.46`+

File: proto/buf.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@ version: v1
 name: buf.build/cosmwasm/wasmd
 deps:
   - buf.build/cosmos/cosmos-proto
-  - buf.build/cosmos/cosmos-sdk
+  - buf.build/cosmos/cosmos-sdk:8cb30a2c4de74dc9bd8d260b1e75e176 #v0.46.x
```
Comment (alpe) [inline proto/buf.yaml:10]: 👌 to use this temporary version

**Review by alpe — APPROVED** [review]
Very nice to see so many files being removed! 👏 

Before squash+merge, would you run `make proto-all` to regenerate the type files, please?

**Pull request merged**